src/test/java/ttt/gui/TicTacToeRulesTest.java

```diff
@@ -78,10 +80,27 @@ public void initialisesGame() {
                 new CommandLinePlayerFactoryStub(players)
         );
         ticTacToeRules.initialiseGame(HUMAN_VS_HUMAN, "3");
+
         assertThat(ticTacToeRules.getBoard(), is(board));
     }
 
     @Test
+    public void currentPlayerReinitialisedWhenNewGameStarted(){
+        Board board = new Board(3);
+
+        TicTacToeRules ticTacToeRules = new TicTacToeRules(
+                new BoardFactoryStub(board, board),
+                new CommandLinePlayerFactoryStub(players)
+        );
+
+        ticTacToeRules.initialiseGame(HUMAN_VS_HUMAN, "3");
+        ticTacToeRules.takeTurn(1);
+        ticTacToeRules.initialiseGame(HUMAN_VS_UNBEATABLE, "3");
+
+        assertThat(ticTacToeRules.getCurrentPlayerIndex(), is(0));
+    }
```

Comment:
The initialisation of TicTacToeRules needs to use the Stubs here, so that a different board can be created each time the boardFactory is called in the initialise method.

I mimic a replay being chosen here, which is when you would call initialiseGame for the second time, hence the test calls initialise game, makes one turn (to toggle the players) then re-initialises the game to ensure the player index is reset.